path-manager/app/controllers/PathManagerController.scala

```diff
@@ -28,37 +28,55 @@ object PathManagerController extends Controller {
   }
 
   def registerExistingPath(id: Long) = Action { request =>
-    request.body.asJson.map(_.as[PathRecord]).map { submission =>
-      if (id != submission.identifier) {
+    val maybeSubmission = request.body.asJson.map(_.as[PathRecord])
+    maybeSubmission match {
+      case (Some(path)) if path.identifier != id => {
         PathOperationErrors.increment
         Logger.warn("registerExistingPath failed, identifier in url and body do not match")
         BadRequest("identifier in url and body do not match")
-      } else if (submission.`type` != "canonical") {
-        PathOperationErrors.increment
-        Logger.warn("registerExistingPath failed, only canonical paths can be updated at present")
-        BadRequest("only canonical paths can be updated at present")
-      } else {
-        PathStore.register(submission) match {
+      }
+
+      case Some(path) if path.`type` == "alias" => {
+        PathStore.registerAlias(path) match {
+          case Left(error) => BadRequest(error)
+          case Right(records) => {
+            PathMigrationRegistrations.increment
+            argoOk(Json.toJson(records))
+          }
+        }
+      }
+      case Some(path) if path.`type` == "canonical"  => {
+        PathStore.registerCanonical(path) match {
           case Left(error) => BadRequest(error)
           case Right(records) => {
             PathMigrationRegistrations.increment
             argoOk(Json.toJson(records))
           }
         }
       }
-    } getOrElse{
-      PathOperationErrors.increment
-      Logger.warn("registerExistingPath failed, unable to parse PathRecord from request body")
-      BadRequest("unable to parse PathRecord from request body")
+      case Some(_)  => {
+        PathOperationErrors.increment
+        Logger.warn("registerExistingPath failed, only canonical and alias paths can be updated at present")
+        BadRequest("only canonical paths can be updated at present")
+      }
+      case None => {
+        PathOperationErrors.increment
+        Logger.warn("registerExistingPath failed, unable to parse PathRecord from request body")
+        BadRequest("unable to parse PathRecord from request body")
+      }
+
     }
-  }
 
-  def updateCanonicalPath(id: Long) = Action { request =>
+  }
 
+  def updateCanonicalPath(id: Long /*, formAlias: Boolean*/) = Action { request =>
     val submission = request.body.asFormUrlEncoded.get
     val newPath = submission("path").head
 
-    PathStore.updateCanonical(newPath, id) match {
+    val update = if (request.getQueryString("formAlias") contains "true") {PathStore.updateCanonicalWithAlias _} else {PathStore.updateCanonical _}
```

Comment:
should this be `fromAlias`?

Reply:
that could be why it's not 'binding'?

Reply:
No, it's an instruction to form an alias from the previous canonical path.

Reply:
I wonder if this is still too close to the `fromAlias` query param you mention in the description etc?

Reply:
in-fact on closer inspection, I think maybe the PR description should read `formAlias`...

Reply:
I also wonder if it should be `shouldFormAlias` just to imply the 'instruction' idea a little more explicitly

Reply:
I think getting it wrong in the PR desc is a good sign that it needs to be better, changed to `shouldFormAlias`